### ISSUE TITLE
[Request] Would like to see OSRIC system

### ISSUE THREAD

**Issue by skolaz** [SkewedAspect/rpgkeeper#146]
Me and my friends are starting a campaign and would like to have an awesome website to host character sheets. I could try to learn some script to help add material.

**Comment by Morgul** [comment]
I'm not opposed to adding this system. (Haven't heard of it before, seems very D&D-esque, which I'm pretty familiar with.)

It appears the rules can be found here: 
* https://osricwiki.presgas.name/doku.php?id=osric:index#preambleosric_-_first_edition_reborn

Typically, simpler systems are easier to get in. This systems appears to have a decent level of complexity, so it'd take more time to add. As RPGKeeper is a hobby project for me (and one I haven't been doing much work with recently) this could take a while to get added.

But! There's things you can to to help, if you really want this. First, I love seeing people getting excited about this project, so that helps me want to work on it. So good job there. 😄  👍 

Second, there's really two major pieces to a system. Figuring out how to store the character for a system (i.e. data model) and then the UI. The UI takes the most work, but can't even be started until the data model is done.

So, what you could do to immediately help get this going is to help me come up with what data needs to be stored for a character. Think of it like using notepad for your character sheet, what would you need to write down? Here's a super simple example:

```markdown
# Character

- name
- ability scores
  - strength
  - dexterity
  - constitution
  - intelligence
  - wisdom
  - charisma
- race
  - ???
- class
  - ???
- ...
  
# Race

- name
- abilities
- languages
- permitted classes
- ...

# Class

- ...
```

In addition, every piece of data RPGK needs to include will need to be modeled. (ex: races, classes, weapons, etc.), so helping to work out what needs to be stored for those, too, would really help.

Once we have a list of what to store, building the data model for that is a few hours/days, depending on how much and how much free time I have. At that point, what you could help with is entering data (for things like classes, racs, etc.) while I work on getting a basic UI up. After that it would be a back/forth with you and your friends as we tweak and test and fix things.

---

TL;DR: This is a lot of work, but if you want it, I'm down to help make it happen. On my own, I probably won't get around to it, but if you're excited and motivated, we can make this happen. :)

**Comment by skolaz** [comment]
Hi! Thanks for your wholesome response. I would gladly do what I can and have time for

**Comment by skolaz** [comment]
Hi! Thanks for your wholesome response. I would gladly do what I can and have time for :)

I accicdently closed this :(

Which shows how new I am to this. I will read through the code and try to copy most of it when I write sheet example from above

**Comment by skolaz** [comment]
Something like this?
```
# Character

- name
- ability scores
  - strength
  - dexterity
  - constitution
  - intelligence
  - wisdom
  - charisma
- race
  -
- class
  -
- alignment
  - Lawful Good
  - Lawful Neutral
  - Lawful Evil
  - Neutral Good
  - Neutral Neutral
  - Neutral Evil
  - Chaotic Good
  - Chaotic Neutral
  - Chaotic Evil
- level
- experience
- deity served
- vision

# Race Dwarf

- name
- abilities
  - +1 constitution, -1 charisma (with respect to all but dwarfs)
  - +1 'to hit' against goblins, half-orcs, hobgoblins, and orcs
  - +1 bonus per 3.5 points of Con to saves against magic
  - -4 penalty to any attacks made against the dwarf by giants, ogres, ogre mages, titans and trolls
- vision
  - Infravision 60-ft
- languages
  - Dwarfish, gnomish, goblin, kobold, orchish, common, alignment tongue
- permitted classes
  - assasin max level 9
  - cleric max level 8
  - fighter max level 9 (strenght 18), max level 8 (strenght 17), max level 7 (strenght 16 or less)
  - thief unlimited
- starting age
  - cleric 250+2d20
  - fighter 40+5d4
  - thief 75+3d6
```

**Comment by skolaz** [comment]
Used sheets from here: https://www.dragonsfoot.org/files/pdf/Milos_OSRIC_Character_Sheet.pdf
And took info from here: https://osricwiki.presgas.name/doku.php?id=osric:chapter1 

Below is a good chuck of what i got so far. Please correct me and give examples on how I can write things correct. With ability scores I tried to do something, but actually have no idea. Really pumped hearing from you again!
```
# character

- name
- ability scores
  - strength
    if (strenght 3) {
      bonus to hit -3
      bonus to damage -1
      encumbrance adjustment -35
      minor tests 1
      major tests 0};
   else if (strenght 4-5) }
      bonus to hit -2
      bonus to damage -1
      encumbrance adjustment -25
      minor tests 1
      minor tests 0
    if (strenght 6-7) }
      bonus to hit -1
      bonus to damage 0
      encumbrance adjustment -15
      minor tests 1
      major tests 0
    if (strenght 8-9) }
      minor tests 1-2
      major tests 1
    if (strenght 10-11) }
      minor tests 1-2
      major tests 2
    if (strenght 12-13) }
      encumbrance adjustment +10
      minor tests 1-2
      major tests 4
    if (strenght 14-15) }
      encumbrance adjustment +20 
      minor tests 1-2
      major tests 7
    if (strenght 16) }
      bonus to damage +1
      encumbrance adjustment +35
      minor tests 1-3
      major tests 10
    if (strenght 17) }
      bonus to hit +1
      bonus to damage +1
      encumbrance adjustment +50
      minor tests 1-3
      major tests 13
    if (strenght 18) }
      bonus to hit +1
      bonus to damage +2
      encumbrance adjustment +75
      minor tests 1-3
      major tests 16
    if (strenght 18.01-18.50) }
      bonus to hit +1
      bonus to damage +3
      encumbrance adjustment +100
      minor tests 1-3
      major tests 20
    if (strenght 18.51-18.75) }
      bonus to hit +2
      bonus to damage +3
      encumbrance adjustment +125
      minor tests 1-4
      major tests 25
    if (strenght 18.76-18.90) }
      bonus to hit +2
      bonus to damage +4
      encumbrance adjustment +150
      minor tests 1-4
      major tests 30
    if (strenght 18.91-18.99) }
      bonus to hit +2
      bonus to damage +5
      encumbrance adjustment +200
      minor tests 1-4
      major tests 35
    if (strenght 19) }
      bonus to hit +3
      bonus to damage +6
      encumbrance adjustment +300
      minor tests 1-5
      major tests 40
  - dexterity
  - constitution
  - intelligence
  - wisdom
  - charisma
- race
  -
- class
  -
- alignment
  - lawful good "crusader"
  - lawful neutral "judge"
  - lawful evil "dominator"
  - neutral good "benefactor"
  - neutral neutral
  - neutral evil "malefactor"
  - chaotic good "rebel"
  - chaotic neutral "free spirit"
  - chaotic evil "destroyer"
- level
- experience
- optional
  - deity served
  - gender
  - hair
  - eyes
  - height
  - weight
- saves
  - aimed magic items
  - breath weapons
  - death, paralyze, poison
  - petrify, polymorph
  - spells
- hit points
  - current
  - max
- vision
- thaco
- equipment
   - weapon
    - #att??
    - to hit
    - damage adjustment
    - damage vs s or m
    - damage vs l
    - range
    - rate of fire
  - armour
    - armour
      - ac
    - shield
      - ac
    - gloves/rings
      - ac
    - bracers
      - ac
    - boots
      - ac
    - cloak
      - ac
  - magic items
  - other
- if class (cleric)
  - cleric turn undead table by type
- if class (thief)
  - climb walls
  - find traps
  - hear noise
  - hide in shadows
  - move quietly
  - open locks
  - pick pockets
  - read languages
- money
  - copper
  - silver
  - electrum
  - gold
  - platinum
- spell book
  - spells per level #number of spells availabe for each spell level. linked to class table. exclusivly for cleric, druid, illusionist, paladin, ranger and magic user
    - 1st
    - 2nd
    - 3rd
    - 4th
    - 5th
    - 6th
    - 7th
    - 8th
    - 9th
  - name
  - level
  - duration
  - range
  - effect
  - page found

# race dwarfs

- name
- abilities #everything available
  - +1 constitution, -1 charisma (with respect to all but dwarfs)
  - +1 'to hit' against goblins, half-orcs, hobgoblins, and orcs
  - +1 bonus per 3.5 points of con to saves against magic
  - -4 penalty to any attacks made against the dwarf by giants, ogres, ogre mages, titans and trolls
  - detect facts concerning engineering, stonework etc.
    - detect the existence of slopes or grades: 75%
    - detect the existence of new construction: 75%
    - detect sliding or shifting rooms or walls: 66%
    - detect traps involving stonework: 50%
    - determine depth underground: 50%
- vision
  - infravision 60-ft
- movement
  - 90-ft
- languages
  - dwarfish, gnomish, goblin, kobold, orchish, common, alignment tongue
- racial limitation #if not between these numbers, character may not be this race
  - strength 8/18
  - dexterity 3/17
  - constitution 12/19
  - intelligence 3/18
  - wisdom 3/18
  - charisma 3/16
- permitted classes
  - assasin max level 9
  - cleric max level 8
  - fighter max level 9 (strenght 18), max level 8 (strenght 17), max level 7 (strenght 16 or less)
  - thief unlimited
- multiclass restrictions
  - the more restrictive of any two class requirements apply to multi-classed dwarfish characters for the use of class abilities.
  - fighter/thief
- starting age
  - cleric 250+2d20
  - fighter 40+5d4
  - thief 75+3d6

# race elves

- name
- abilities #everything available
  - +1 dexterity, -1 constitution
  - 90% resistance to sleep and charm spells
  - any pulled bow: +1 "to hit"
  - longsword and short sword: +1 "to hit"
  - detect secret doors
  - suprise
- vision
  - infravision 60-ft
- movement
  - 120-ft
- languages
  - common, elven, gnoll, gnomish, goblin, halfling, hobgoblin, orcish
- racial limitation #if not between these numbers, character may not be this race
  - strength 3/18
  - dexterity 7/19
  - constitution 8/17
  - intelligence 8/18
  - wisdom 3/18
  - charisma 8/18
- permitted classes
  - assasin max level 10
  - cleric max level 7
  - fighter max level 7 (strenght 18), max level 6 (strenght 17), max level 5 (strenght 16 or less)
  - magic user max level 11 (intelligens 18 or more), max level 10 (intelligens 17), max level 9 (intelligens 16 or less)
  - thief unlimited
- multiclass restrictions
  - the less restrictive of any two class requirements apply to multi-classed elven characters, except that thieving abilities can only be used while wearing armour permitted to thieves.
  - fighter/magic user, fighter/thief, magic user/thief, fighter/ magic user/thief.
- starting age
  - cleric 500+10d10
  - fighter 130+5d6
  - magic user 150+5d6
  - thief 100+5d6

# race gnomes

- name
- abilities #everything available
  - +1 bonus per 3.5 points of con to saves against magic and poison
  - +1 "to hit" kobolds and goblins
  - -4 to attack rolls by bugbears, giants, gnolls, ogres, ogre mages, titans, and trolls.
  - detect certain facts concerning engineerin, stonework etc.
    - detect the existence of slopes or grades: 80%
    - detect the existence of unsafe wall, ceiling, floor: 70%
    - determine depth underground: 60%
    - determine direction of north underground: 50%
- vision
  - infravision 60-ft
- movement
  - 90-ft
- languages
  - common, dwarfish, gnomish, goblin, halfling, kobold, any normal burrowing animal
- racial limitation #if not between these numbers, character may not be this race
  - strength 6/18
  - dexterity 3/18
  - constitution 8/18
  - intelligence 7/18
  - wisdom 3/18
  - charisma 3/18
- permitted classes
  - assasin max level 8
  - cleric max level 7
  - fighter max level 6 (strenght 18), max level 5 (strenght 17 or less)
  - illusionist max level 7 (one of dex or intelligens 18+ and the other 17+), max level 6 (dex and intelligens 17), max level 5(dex and intelligens 16 or less)
  - thief unlimited
- multiclass restrictions
  - only leather armour, no better
  - fighter/illusionist, fighter/thief, illusionist/thief.
- starting age
  - cleric 300+3d12
  - fighter 60+5d4
  - magic user 100+2d12
  - thief 80+5d4

# race half elves

- name
- abilities #everything available
  - 30% resistance to sleep and charm spells
  - secret doors: 1-2 on 2d6 detect secret doors. concealed doors show on 1-3 on d6. when passing within 10-ft of a concealed door, a half-elf will notice it on a 1 in 6.
- vision
  - infravision 60-ft
- movement
  - 120-ft
- languages
  - common, elven, gnoll, gnome, goblin, halfling, hobgoblin, orcish
- racial limitation #if not between these numbers, character may not be this race
  - strength 6/18
  - dexterity 3/18
  - constitution 8/18
  - intelligence 7/18
  - wisdom 3/18
  - charisma 3/18
- permitted classes
  - assasin max level 8
  - cleric max level 5
  - fighter max level 8 (strenght 18), max level 7 (strenght 17 or less), max level 6 (strenght 16 or less)
  - magic user max level 8 (intelligens 18+), max level 7 (intelligens 17), max level 6 (intelligens 16 or less)
  - ranger max level 8 (strenght 18), max level 7 (strenght 17 or less), max level 6 (strenght 16 or less)
  - thief unlimited
- multiclass restrictions
  - the less restrictive of any class requirements apply to multi-classed half-elven characters, except that thieving abilities can only be used while wearing armour permitted to thieves.
  - cleric/fighter, cleric/ranger, cleric/magic user, fighter/magic user, fighter/thief, magic user/thief, cleric/fighter/ magic user, or fighter/magic user/thief.
- starting age
  - cleric 40+2d4
  - fighter 22+3d4
  - magic user 30+2d8
  - thief 22+3d8

# race halflings

- name
- abilities #everything available
  - -1 str, +1 dex
  - +1 bonus per 3.5 points of con to saves against magic (both aimed magic items and spells) and poison.
  - +3 bonus to attacks with a bow or sling
  - suprise
- vision
  - infravision 60-ft
- movement
  - 90-ft
- languages
  - common, dwarfish, gnome, goblin, halfling and orcish
- racial limitation #if not between these numbers, character may not be this race
  - strength 6/17
  - dexterity 8/19
  - constitution 10/18
  - intelligence 6/18
  - wisdom 3/17
  - charisma 3/18
- permitted classes
  - druid max level 6
  - fighter max level 4
  - thief unlimited
- multiclass restrictions
  - use of thieving abilities is only possible when wearing armour permitted to thieves.
  - fighter/thief.
- starting age
  - fighter 20+3d4
  - druid 40+3d4
  - thief 40+2d4

# race half orcs

- name
- abilities #everything available
  - +1 str and con, -2 cha
- vision
  - infravision 60-ft
- movement
  - 120-ft
- languages
  - common, orcish
- racial limitation #if not between these numbers, character may not be this race
  - strength 6/18
  - dexterity 3/17
  - constitution 13/19
  - intelligence 3/17
  - wisdom 3/14
  - charisma 3/12
- permitted classes
  - assasin max level 15
  - cleric max level 4
  - fighter max level 10
  - thief max level 7 (dex 17), max level 6 (dex 16 or less)
- multiclass restrictions
  - for armour, the more restrictive of any two class requirements apply to multi-classed half orc characters for the use of class abilities. for weapons, the less restrictive requirements apply.
  - cleric/fighter, cleric/thief, cleric/assassin, fighter/thief, fighter/ assassin.
- starting age
  - cleric 20+1d4
  - fighter 13+1d4
  - thief 20+2d4

# race humans

- name
- abilities
  - 
- vision
  - 
- movement
  - 120-ft
- languages
  - Common
- racial limitation #if not between these numbers, character may not be this race
  - 
- permitted classes
  - assassin max level 15
  - cleric unlimited
  - druid max level 14
  - fighter unlimited
  - illusionist unlimited
  - magic user unlimited
  - paladin unlimited
  - ranger unlimited
  - thief unlimited
- multiclass restrictions
  -
- starting age
  - cleric 20+1d4
  - fighter 15+1d4
  - mage 24+2d8
  - thief 20+1d4

# class
- assasin
- cleric
- druid
- fighter
- illusionist
- magic user
- paladin
- ranger
- thief
```

**Comment by skolaz** [comment]
I have compiled alot. It was fun! Instead of sending a lot of rows here I am sharing the # character via OneDrive. I am waiting for response if I am doing right. Otherwise I continue the way I written it so far

https://onedrive.live.com/?authkey=%21AMa18t4CBXvjG6Y&cid=E3B46065A7763E89&id=E3B46065A7763E89%214499&parId=E3B46065A7763E89%214225&o=OneUp
